official/nlp/modeling/ops/beam_search.py

```diff
@@ -455,19 +453,19 @@ def _create_initial_state(self, initial_ids, initial_cache, batch_size):
               tf.TensorShape([]),
           _StateKeys.ALIVE_SEQ:
               tf.TensorShape(
-                  [batch_size, self.beam_size, self.max_decode_length + 1]),
```

Comment:
The batch size need to be "known" at least for TPUs.

Reply:
yes that's true, but so far it don't work either on classical hardware.

Reply:
I am pretty sure this works on TPU/GPU savedmodel serving. The beam search code is used in multiple projects. If this change is useful to your ONNX path, you can keep in your fork.